### PR TITLE
V2.0.0 kohai 206 check for current daemon on users system

### PR DESCRIPTION
**Pull request by red010b37**

### CODE REVIEW

**Comment by red010b37** [comment]
Please note only the windows daemon is supported at this stage.

File: app/daemon/daemon.go [hunk 1]
```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
...
 	log.Println("Looking for Navcoin Daemon at " + path )
 
-
 	// check the daemon exists
 	if !exists(path) {
 		log.Println("No Daemon found for version: " + releaseVersion)
-
-		return true, nil
+		return "", errors.New("No Daemon found for version: " + releaseVersion)
 	}else {
```
Comment (matt-auckland) [inline app/daemon/daemon.go]: There are a few places where you use `} else {` and in other places (like here) you use `}else {` 
Pretty anal but, it's nice to keep it consistent. Wonder if we could get a Go Linter running? (though from what I understand the compiler does some linting of it's own)

File: app/daemon/daemon.go [hunk 2]
```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
...
+	return e, err
 
 }
 
 
-func getDaemonDownloadPath(version string) {
+func gitHubReleaseInfo(releaseAPI string) (GitHubReleases, error) {
 
-	
+	log.Println("Retreving NAVCoin Github releases info from: " + releaseAPI)
```
Comment (matt-auckland) [inline app/daemon/daemon.go]: I think it might be better as `Retrieving NAVCoin Github releases' info from:`

File: app/daemon/daemon.go [hunk 3]
```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
...
+	return c, nil
 }
 
-func downloadDaemon(config *conf.ServerConfig, version string) error {
 
-	log.Println("Updating Daemon")
-	path, err := getCurrentPath()
+func getDwLdInfoFromReleaseInfo(gitHubReleaseData GitHubReleaseData) (string, string, error) {
+
+	log.Println("Getting dowload path for OS from release assest data")
```
Comment (matt-auckland) [inline app/daemon/daemon.go]: Missing an n in download here

File: app/daemon/daemon.go [hunk 4]
```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
...
 
-	log.Println("Latest version: " + releaseVersion)
+	log.Println("Looking for version: " + releaseVersion)
 
 	// get the apps current path
 	path, err := getCurrentPath()
 	if err != nil {
-		return false, err
+		return "", err
 	}
```
Comment (matt-auckland) [inline app/daemon/daemon.go]: No issue here, just wanted to say that being able to return multiple things at once is sweeeeet

Comment (red010b37) [reply app/daemon/daemon.go]: So much win!

File: app/daemon/daemon.go (hunk 2 above)
Comment (red010b37) [reply app/daemon/daemon.go]: done :)

File: app/daemon/daemon.go (hunk 1 above)
Comment (red010b37) [reply app/daemon/daemon.go]: fixed that instance, and i agree with you.

 You are right go has one build in but it doesn't catch things like that.  I will see what i can find for a linter.

File: app/daemon/daemon.go (hunk 3 above)
Comment (red010b37) [reply app/daemon/daemon.go]: fixed

**Pull request merged**